packages/flutter/lib/src/material/pickers/calendar_date_range_picker.dart

```diff
@@ -110,7 +110,8 @@ class _CalendarDateRangePickerState extends State<CalendarDateRangePicker> {
     // divide the list of months into two `SliverList`s.
     final DateTime initialDate = widget.initialStartDate ?? widget.currentDate;
     if (widget.firstDate.isBefore(initialDate) &&
-        widget.lastDate.isAfter(initialDate)) {
+        widget.lastDate.isAfter(initialDate) ||
```

Comment:
Nice catch. However, while we don't see the same problem with `firstDate` here, that is only because it `_initialMonthIndex` defaults to 0. I would still like to see both cases handled here so that it is obvious both are handled. Perhaps something like:

```dart
    if (!initialDate.isBefore(widget.firstDate) && 
        !initialDate.isAfter(widget.lastDate)) {
      _initialMonthIndex = utils.monthDelta(widget.firstDate, initialDate);
    }
```

Reply:
Oh this is much better! Thank you.